### PR TITLE
Include the python version being used in the user agent for ACME requ…

### PR DESCRIPTION
**Pull request by alex**
…ests.

### CODE REVIEW

**Comment by alex** [comment]
cc: @reaperhulk

**Comment by pde** [comment]
Hi @alex; is there a specific reason why we need to collect this?

**Comment by alex** [comment]
@pde the local impetus is that pyca/cryptography is interested in deprecating Python 2.6 support, letsencrypt is the downstream most likely to be affected, and we'd like to collect data to see how many folks would be affected.

The more general reason is that it's a useful thing to have data on :-)

File: acme/acme/client.py [hunk 1]
```diff
@@ -483,6 +484,8 @@ def revoke(self, cert):
                 'Successful revocation must return HTTP OK status')
 
 
+DEFAULT_USER_AGENT = 'acme-python Python/%s' % platform.python_version()
+
```
Comment (kuba) [inline acme/acme/client.py]: there should be 2 separating newlines between constant and class here

**Comment by kuba** [comment]
I'm not sure if we should disclose user's Python version, at least in the reusable `acme` library by default. That gives an attacker more info about the target operating system. Core client already discloses a ton of data, so maybe let's keep the change scope to `letsencrypt/` dir and revert `acme`?

**Comment by alex** [comment]
@kuba done.

**Comment by kuba** [comment]
Thanks, @alex :)

**Comment by pde** [comment]
@alex [these](https://docs.google.com/spreadsheets/d/1i0O9EO61NL-YyAHp0OlOXk-5GliMwfO1qrEr5Gx01ek/edit#gid=1767154651) are the top OSes the client is running on at present.

I think this implies that a bit over 10% of clients are running on platforms where Python 2.6 is the default.  Does that sufficiently answer your question?

**Comment by pde** [comment]
Actually maybe more like 13-15%, a large fraction of "Other" is older OSes including Centos 6 variants and  Debian 6.

**Comment by alex** [comment]
@pde thanks!

**Comment by pde** [comment]
Closing this for now then.

**Pull request closed without merging**